### ISSUE TITLE
Purging episode data in get_episodes.py if connection fails

### ISSUE THREAD

**Issue by mortenmhp** [morpheus65535/bazarr#416]
This is a repost of issue #414 Sorry for being a bit insisting here, but I you closed my previous issue without addressing it at all. 

Increasing the timeout doesn't fix the issue(I knew about the change in the dev branch already), as that wasn't the issue raised. The issue is that it doesn't make sense to delete all episode data for a show simply because the connection to sonarr fails doing the sync of episodes for whatever reason, not just because of timeout, forcing bazarr to rescan every file on the disk on the next successful sync.

If you close this as well I'll take the hint, but I'd like that you address the actual issue not just mitigate one cause.

**Original:**
**Describe the bug**
I'm running a fairly slow server for sonarr/bazarr meaning that sometimes sonarr can't respond with the episode list for a series within 15s. I've increased the timeout manually, but it resets at every update and every now and then a connection error happens anyway. The issue is that whenever an exception is thrown for the request to get the episode list from sonarr, the episodes won't be added to current_episodes_sonarr and therefore they'll end up on the list of episodes to be deleted from the DB. This means that the next time the request goes through, the DB has to be rebuilt from the ground for that show. This seems unreasonable.

**To Reproduce**
Throw exception in get_episodes.py during the request to the episodes API in sonarr.

**Solution**
A solution could be to move the addition of episodes to the deleted_episodes list inside the "for seriesId in seriesIdList:" for loop inside the else clause of the try/except block to do it on a per series basis only if the request goes through. The calculation would obviously have to be modified as well.
If this is the only way episode data is deleted(e.g. after a show is deleted in sonarr), it may need to be done slightly differently if sonarrs response throws an exception for an unknown series, but it seems like it could be done quite easily.

**Comment by morpheus65535** [comment]
I agree and understand your point. My apologies for closing 414 without addressing it completely. I’ll look into this.
